api/commands/run.ts

```diff
@@ -58,15 +58,22 @@ export class Runner {
     if (!!this.executionTask) {
       throw Error("Executor already started.");
     }
-    const prepareDefaultSchema = this.adapter.prepareSchema(this.graph.projectConfig.defaultSchema);
-    const prepareAssertionSchema = this.adapter.prepareSchema(
-      this.graph.projectConfig.assertionSchema
+
+    // Work out all the schemas we are going to need to create first.
+    const uniqueSchemas = {};
+    this.graph.nodes
+      .filter(node => !!node.target)
+      .map(node => node.target.schema)
+      .filter(schema => !!schema)
+      .forEach(schema => (uniqueSchemas[schema] = true));
+
+    const prepareSchemas = Promise.all(
```

Comment:
since you don't actually use this in this function, suggest pushing `prepareSchemas` into the Promise function below, probably just inline it where it's used.